src/components/HamburguerProfileMenu/index.tsx

```diff
@@ -0,0 +1,53 @@
+'use client';
+
+import { Menu } from '@/styles/Icons';
+import { useState } from 'react';
+import PrimaryButton from '../PrimaryButton';
+import LogoutButton from '../LogoutButton';
+
+interface HamburgerProfileMenuProps {
+  token: string | undefined;
+  isMobile?: boolean;
+}
+
+const HamburgerProfileMenu: React.FC<HamburgerProfileMenuProps> = ({ token, isMobile }) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleClickMenu = () => {
+    setIsOpen((prev) => !prev);
+  };
+
+  return (
+    <div className="flex">
+      <Menu className="text-primary hover:cursor-pointer" onClick={handleClickMenu} />
+      {isOpen && (
+        <div
+          className={`absolute w-50 bg-white rounded shadow ${
```

Comment:
the alignment on mobile is weird too maybe in mobile don't make it a submenu and just separate both options with a separator.
Plus, when navigating to some components (I think server ones but I need to be sure) hamburger menu doesn't collapse automaticly

Reply:
@tomasflopes 
The mobile menu is only colapsed , when the root layout is refresh.
This happens when you redirect the user to the root page, thats why sometimes it closes the the hamburguer menu (because the component is refresh) and sometimes not.
We need to rethink our approach